portfolio/src/main/java/com/google/sps/servlets/LikeCommentServlet.java

```diff
@@ -0,0 +1,129 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.*;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+
+@WebServlet("/toggle-like")
+public class LikeCommentServlet extends HttpServlet {
+
+  private DatastoreService datastore;
+
+  @Override
+  public void init() {
+    datastore = DatastoreServiceFactory.getDatastoreService();
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String commentKeyString = request.getParameter("key");
+    Entity comment = null;
+    try {
+      comment = datastore.get(KeyFactory.stringToKey(commentKeyString));
+    } catch (EntityNotFoundException e) {
+      e.printStackTrace();
+      response.sendRedirect("/");
+      return;
+    }
+
+    // The first element in this list records whether or not the client has like this comment, and the second records the number of likes this comment has.
+    List<String> commentLikeData = new LinkedList<>();
```

Comment:
I think defining a simple class for this would be better for structuring data. The output json will be better structured and clearer.